discord/primary_guild.py

```diff
@@ -0,0 +1,75 @@
+"""
+The MIT License (MIT)
+
+Copyright (c) 2015-present Rapptz
+
+Permission is hereby granted, free of charge, to any person obtaining a
+copy of this software and associated documentation files (the "Software"),
+to deal in the Software without restriction, including without limitation
+the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the
+Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+"""
+
+from __future__ import annotations
+
+from typing import Optional, TYPE_CHECKING
+from datetime import datetime
+
+from .asset import Asset
+from .utils import snowflake_time, _get_as_snowflake
+
+if TYPE_CHECKING:
+    from .state import ConnectionState
+    from .types.primary_guild import PrimaryGuild as PrimaryGuildPayload
+
+
+class PrimaryGuild:
+    """Represents the primary guild (formally known as a clan) of a :class:`User`"""
+
+    __slots__ = ('id', 'identity_enabled', 'tag', '_badge', '_state')
+
+    if TYPE_CHECKING:
```

Comment:
what's the point of the type checking block here? just have the types inline (also `identity_enabled` and `_badge` are both optional)